bitcoin/src/blockdata/transaction.rs

```diff
@@ -53,21 +53,29 @@ impl_hashencode!(Txid);
 impl_hashencode!(Wtxid);
 
 impl Txid {
-    /// The "all zeros" TXID.
+    /// The `Txid` used in a coinbase prevout.
     ///
     /// This is used as the "txid" of the dummy input of a coinbase transaction. It is
     /// not a real TXID and should not be used in other contexts.
-    pub fn all_zeros() -> Self { Self::from_byte_array([0; 32]) }
+    pub const COINBASE_PREVOUT: Self = Self::from_byte_array([0; 32]);
+
+    /// The "all zeros" TXID.
+    #[deprecated(since = "TBD", note = "use Txid::COINBASE_PREVOUT instead")]
+    pub fn all_zeros() -> Self { Self::COINBASE_PREVOUT }
 }
 
 impl Wtxid {
-    /// The "all zeros" wTXID.
+    /// The `Wtxid` of a coinbase transaction.
     ///
     /// This is used as the wTXID for the coinbase transaction when constructing blocks,
     /// since the coinbase transaction contains a commitment to all transactions' wTXIDs
     /// but naturally cannot commit to its own. It is not a real wTXID and should not be
     /// used in other contexts.
-    pub fn all_zeros() -> Self { Self::from_byte_array([0; 32]) }
+    pub const COINBASE: Self = Self::from_byte_array([0; 32]);
```

Comment:
Why is this one named differently?

Reply:
Because it has a different meaning.

Reply:
Oh, right, the field is not called *prevout* but there should be a suffix to  not confuse it with "wtxid of coinbase".

Reply:
It *is* the "wtxid of the coinbase transaction" though.

Reply:
There is an interesting question as to whether our `compute_wtxid` method should look for coinbase inputs and return this constant. There are two places that wtxids are used:

* To identify transactions in the mempool/p2p in a way that includes their witness data (irrelevant, coinbases are illegal in these contexts)
* To build the witness merkle root in the blockheader (in which case the coinbase tx gets the all-zeroes wtxid).

So it sounds like we *should* return the "wtxid of a coinbase transaction" as this constant value  ... but I worry that this would be surprising or even dangerous if users were trying to use the wtxid as a collision-free transaction identifier in other contexts.

Reply:
Oh, interesting. So it looks like wTXID is only intended for commitment or in contexts where coinbases are not allowed - it must not be used for anything else. And if our `Wtxid` type == wTXID (very reasonable) then we should somehow reflect this. Perhaps we should have two methods:

* `compute_wtxid_for_commitment` - used to build the commitment, contains coinbase detection you suggested
* `compute_wtxid_for_non_coinbase_identification` - the current `compute_wtxid`

We could also have two types or `Wtxid<Purpose>` where purpose is either `Commitment` or `NonCoinbaseIdentifier` but I think it's too much because commitment computation is a single algorithm which we implement anyway and can handle correctly.

Reply:
Added issue to track this: #3313